lib/Conch/Controller/User.pm

```diff
@@ -19,31 +19,36 @@ Conch::Controller::User
 
 =head1 METHODS
 
-=head2 revoke_own_tokens
+=head2 revoke_user_tokens
 
-Revoke all the user's own session tokens.
-B<NOTE>: This will cause the next request to fail authentication.
+Revoke a specified user's tokens and prevents future token authentication,
+forcing the user to /login again. By default *all* of a user's tokens are deleted,
+but this can be adjusted with query parameters:
 
-=cut
+ * ?login_only=1    login tokens are removed; api tokens are left alone
+ * ?api_only=1      login tokens are left alone; api tokens are removed
 
-sub revoke_own_tokens ($c) {
-	$c->log->debug('revoking user token for user ' . $c->stash('user')->name . ' at their request');
-	$c->stash('user')->delete_related('user_session_tokens');
-	$c->status(204);
-}
+If login tokens are affected, C<user_session_auth> is also set for the user, which forces the
+user to change his password as soon as a login token is used again (but use of any existing api
+tokens is allowed).
 
-=head2 revoke_user_tokens
-
-Revoke *all* of a specified user's session tokens. System admin only.
+System admin only (unless reached via /user/me).
 
 =cut
 
 sub revoke_user_tokens ($c) {
 	my $user = $c->stash('target_user');
 
+    return $c->status(400, { error => 'cannot use login_only and api_only together' })
```

Comment:
The default behavior is equivalent  to `login_only=1` combined with `api_only=1`, yes? So while it's silly to provide them both, why bother throwing an error?

Reply:
No, combining "login only" AND "api only" is the empty set, as would be `?login_only=1&api_only=1`. The default is `login_only=0&api_only=0`.